components/PlantForm/PlantForm.js

```diff
@@ -85,6 +107,39 @@ export default function PlantForm({
     return;
   }
 
+  function handleInputChange(event) {
+    const { name, value } = event.target;
+    setErrors((prevErrors) => {
+      const newErrors = { ...prevErrors };
+      if (value) {
+        delete newErrors[name];
+      }
+      return newErrors;
+    });
+  }
+
+  function handleRadioChange(event) {
```

Comment:
Very good logic for custom error validation on radio buttons. But you might not need it all if you change the way the radio buttons are checked by default. For details please refer to my comment on line 146.